boards/pygamer/examples/neopixel_rainbow_spi.rs

```diff
@@ -31,32 +31,20 @@ fn main() -> ! {
         &mut peripherals.OSCCTRL,
         &mut peripherals.NVMCTRL,
     );
-    let mut pins = Pins::new(peripherals.PORT);
+    let pins = Pins::new(peripherals.PORT);
+    let mclk = &mut peripherals.MCLK;
     let gclk = clocks.gclk0();
-
-    let spi: hal::sercom::SPIMaster2<
-        hal::sercom::Sercom2Pad0<hal::gpio::Pa12<hal::gpio::PfC>>,
-        hal::sercom::Sercom2Pad3<hal::gpio::Pa15<hal::gpio::PfC>>,
-        hal::sercom::Sercom2Pad1<hal::gpio::Pa13<hal::gpio::PfC>>,
-    > = hal::sercom::SPIMaster2::new(
-        &clocks.sercom2_core(&gclk).unwrap(),
-        MegaHertz(3),
-        embedded_hal::spi::Mode {
-            phase: embedded_hal::spi::Phase::CaptureOnFirstTransition,
-            polarity: embedded_hal::spi::Polarity::IdleLow,
-        },
-        peripherals.SERCOM2,
-        &mut peripherals.MCLK,
-        (
-            pins.sda.into_pad(&mut pins.port),
-            pins.neopixel.into_pad(&mut pins.port),
-            pins.scl.into_pad(&mut pins.port),
-        ),
-    );
-
+    let clock = &clocks.sercom2_core(&gclk).unwrap();
+    let sercom2 = peripherals.SERCOM2;
+    let pads = spi::Pads::default()
+        .data_in(pins.sda)
+        .data_out(pins.neopixel)
+        .sclk(pins.scl);
+    let spi = spi::Config::new(mclk, sercom2, pads, clock.freq())
+        .baud(3.mhz())
+        .enable();
```

Comment:
I think this is a great example of the benefits of the sercom::v2::spi interface. Half the code for the same thing, and the new version is much clearer, IMO, within all types inferred.